rest-api/src/controllers/ImageController.ts

```diff
@@ -7,411 +13,471 @@ import { ImageService } from "../client/images/ImageService";
-                        throw new RequestError(404, "Image not found");
-                    case LikeImageResponseStatus.USER_NOT_FOUND_ERROR:
-                        throw new RequestError(404, "User not found");
-                    default:
-                        ctx.log.warn(e);
-                        throw new RequestError(500);
-                }
-            }
-            ctx.log.error(e);
-            throw new RequestError(503);
+  private postComment = async (
+      ctx: Router.RouterContext<AuthState & IValidationState>) => {
+    const results = validationResults(ctx as any);
+    if (results.hasErrors()) {
+      throw new RequestError(422, { errors: results.array() });
+    }
+    const { id } = ctx.state.account;
+    const { imageId } = ctx.params;
+    const { content, tags, userTags } = ctx.request.body;
+    try {
+      const response = await this.commentService.createComment(
+        content,
+        id,
+        imageId,
+        tags,
+        userTags
+      );
+      ctx.body = response;
+      ctx.status = 201;
+    } catch (e) {
+      if (e instanceof CreateCommentError) {
+        switch (e.status) {
+          case CreateCommentErrorStatus.CREATE_INVALID_PARAMETER:
+            throw new RequestError(400, "Invalid parameters");
+          case CreateCommentErrorStatus.CREATE_SERVER_ERROR:
+          default:
+            ctx.log.warn(e);
+            throw new RequestError(500);
         }
+      }
+      ctx.log.error(e);
+      throw new RequestError(503);
     }
+  };
 
-    /**
-     * Get image likes.
-     */
-    private getLikes = async (
-        ctx: RouterContext<IValidationState>
-    ) => {
-        const results = validationResults(ctx);
-        if (results.hasErrors()) {
-            throw new RequestError(422, { errors: results.array() });
-        }
-        const { imageId, page, size } = results.passedData();
-        const parsedPage = page ? parseInt(page, 10) : 1;
-        const parsedSize = size ? parseInt(size, 10) : 10;
-        try {
-            const likes = await this.imageService.getLikes(
-                imageId,
-                parsedPage,
-                parsedSize
-            );
-            ctx.body = likes;
-        } catch (e) {
-            if (e instanceof GetLikesError) {
-                switch (e.reason) {
-                    case GetLikesErrorStatus.GET_LIKES_IMAGE_NOT_FOUND:
-                        throw new RequestError(404, "Image not found");
-                    default:
-                        ctx.log.warn(e);
-                        throw new RequestError(500);
-                }
+  private getComments = async (ctx: Router.RouterContext<IValidationState>) => {
+    const results = validationResults(ctx);
+    if (results.hasErrors()) {
+      throw new RequestError(422, { errors: results.array() });
+    }
+    const { imageId } = ctx.params;
+    const { page, size } = results.passedData();
```

Comment:
I don't think this works as there are no validations declared for the route. The `results.passedData()` will return data that has been successfully validated (and sanitized). You can just use `this.pageValidation` for this route in order to validate `page` and `size` query parameters - otherwise you'll have to get those from `ctx.query`.